censys_connector.py

```diff
@@ -14,127 +14,24 @@
 # and limitations under the License.
 #
 #
-import ipaddress
 import json
-import math
-import time
 
 import phantom.app as phantom
 import requests
 from phantom.action_result import ActionResult
 from phantom.base_connector import BaseConnector
 
 from censys_consts import *
+from censys_search import CensysSearch
+from censys_validation import get_error_message_from_exception, is_ip, parse_http_error
 
 
 class CensysConnector(BaseConnector):
     def __init__(self):
         self._headers = {}
+        self.search = CensysSearch(self._make_rest_call)
         super().__init__()
```

Comment:
I am used to enabling the dependency injection when possible, so maybe we can consider it here too:
```suggestion
    def __init__(self, censys_search: CensysSearch | None = None):
        self._headers = {}
        self.search = censys_search if censys_search else CensysSearch(self._make_rest_call)
        super().__init__()
```

Just the suggestion tho.

Reply:
IMHO an overkill, chances of us needing more than one "censys_search" kind of object are very low

Reply:
Maybe overkill but no cost update with potential gain

Reply:
And also it makes it possible to have for example CensysSearch for the old censys API, current API and some possible future version of API and just switch them with one env. Who knows what will happen with this connector in the future

Reply:
1) YAGNI
2) The code with dependency injection is more verbose and less readable, which would be understendable if it were needed, but it's not

Reply:
It's a simple change to make, but as mentioned we don't need this now.
When we'll have to cover support for multiple APIs we'll use this pattern